app/controllers/wikis_controller.rb

```diff
@@ -42,11 +48,19 @@ def destroy
   private
 
   def wikis
-    @wikis ||= Wiki.all
+    if current_user.premium? || current_user.admin?
+      @wikis ||= Wiki.all
+    else
+      @wikis ||= Wiki.where(private: nil)
+    end
   end
 
   def wiki
     @wiki ||= Wiki.find(params[:id])
+    # unless @wiki.readable_by?(current_user)
+    #   redirect_to wikis_path, alert: 'Update to Premium in order to view private wikis.'
```

Comment:
You can't use `redirect_to` here because that is a controller specific method and needs to happen before a view gets rendered. This method gets called in the views, hence the error you were seeing.

Putting it in the action above is a better way.